crates/cext/src/transpiler/passes/elide_permutations.rs

```diff
@@ -0,0 +1,195 @@
+    result: *const ElidePermutationsResult,
+) -> bool {
+    let result = unsafe { const_ptr_as_ref(result) };
+    result.elided
+}
+
+/// @ingroup QkElidePermutationsResult
+/// Get the circuit from the elide permutations result
+///
+/// @param result a pointer to the result of the pass. It must have elided gates as checked by
+/// ``qk_elide_permutations_result_elided_gates()``
+///
+/// @returns A pointer to a circuit with the elide
+///
+/// # Safety
+///
+/// Behavior is undefined if ``result`` is not a valid, non-null pointer to a
+/// ``QkVF2LayoutResult``. The pointer to the returned circuit is owned by the
+/// result object, it should not be passed to ``qk_circuit_free()`` as it will
+/// be freed by ``qk_elide_permutations_result_free()``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_elide_permutations_result_circuit(
+    result: *const ElidePermutationsResult,
+) -> *const CircuitData {
+    let result = unsafe { const_ptr_as_ref(result) };
+    result
+        .circuit
+        .as_ref()
+        .map(|x| x as *const _)
+        .expect("Result didn't elide any gates")
+}
+
+/// @ingroup QkElidePermutationsResult
+/// Get the permutation array for the elided gates
+///
+/// @param result a pointer to the result of the pass. It must have elided gates as checked by
+/// ``qk_elide_permutations_result_elided_gates()``
+///
+/// @returns A pointer to the permutation array caused by the swap elision performed by the
+/// pass. This array has a length equal to the number of qubits of the circuit returned by
+/// ``qk_elide_permutations_result_elided_gates()`` (or the circuit passed into
+/// ``qk_transpiler_pass_standalone_elide_permutations()``). The permutation array maps the
+/// virtual qubit in the original circuit at each index to its new output position after all
+/// the elision performed by the pass. For example, and array of ``[2, 1, 0]`` means that
+/// qubit 0 is now in qubit 2 on the output of the circuit and qubit 2's is now at 0 (qubit 1
+/// remains unchanged)
+///
+/// # Safety
+///
+/// Behavior is undefined if ``layout`` is not a valid, non-null pointer to a
+/// ``QkVF2LayoutResult``. Also qubit must be a valid qubit for the circuit and
+/// there must be a result found. The pointer to the permutation array is owned by the
+/// result object, it should not be passed to ``free()`` as it will be freed by
+/// ``qk_elide_permutations_result_free()`` when that is called.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_elide_permutations_result_permutation(
+    result: *const ElidePermutationsResult,
+) -> *const usize {
+    let result = unsafe { const_ptr_as_ref(result) };
+    result
+        .permutation
+        .as_ref()
+        .map(|x| x.as_ptr())
+        .expect("Result didn't elide any gates")
+}
+
+/// @ingroup QkVF2LayoutResult
+/// Free a ``QkVF2LayoutResult`` object
+///
+/// @param layout a pointer to the layout to free
```

Comment:
```suggestion
/// @param result a pointer to the layout result to free
```

Reply:
I adjust this slightly because there was more wrong in the docstring and applied it already.